sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala

```diff
@@ -253,38 +414,65 @@ trait Interpreter extends ScorexLogging {
       val initCost = Evaluation.addCostChecked(context.initCost, complexityCost, context.costLimit)
       val contextWithCost = context.withInitCost(initCost).asInstanceOf[CTX]
 
-      val res = fullReduction(ergoTree, contextWithCost, env)
+      val (aotReduced, jitReduced) = fullReduction(ergoTree, contextWithCost, env)
+
+      // if necessary perform verification as v4.x (AOT based implementation)
+      var aotRes: VerificationResult = null
+      if (evalSettings.evaluationMode == AotEvaluationMode ||
+          evalSettings.evaluationMode == TestEvaluationMode) {
+          aotRes = aotReduced.value match {
+            case TrivialProp.TrueProp => (true, aotReduced.cost)
+            case TrivialProp.FalseProp => (false, aotReduced.cost)
+            case _ =>
+              val ok = if (evalSettings.isMeasureOperationTime) {
+                val E = ErgoTreeEvaluator.forProfiling(verifySignatureProfiler, evalSettings)
+                verifySignature(aotReduced.value, message, proof)(E)
+              } else {
+                verifySignature(aotReduced.value, message, proof)(null)
+              }
+              (ok, aotReduced.cost)
+          }
+      }
 
-      val checkingResult = res.value match {
-        case TrivialProp.TrueProp => true
-        case TrivialProp.FalseProp => false
-        case cProp => verifySignature(cProp, message, proof)
+      // if necessary perform verification as v5.x (JIT based implementation)
+      var jitRes: VerificationResult = null
+      if (evalSettings.evaluationMode == AotEvaluationMode ||
```

Comment:
should be *evalSettings.evaluationMode == AotEvaluationMode* I guess

Reply:
No, this is boolean OR operation.

Reply:
Ah, yes, good catch, fixed.

Reply:
I guess tests are needed, weird that wasn't caught by any test before

Reply:
Fixed, some tests (which check new costs) in SigmaDslSpecification will fail without the fix.